lib/js-file.js

```diff
@@ -134,6 +136,50 @@ JsFile.prototype = {
     },
 
     /**
+     * Sets whitespace before specified token.
+     *
+     * @param {Object} token
+     * @param {String} whitespace
+     */
+    setWhitespaceBefore: function(token, whitespace) {
```

Comment:
There is two primary code paths - if we have whitespace and when we need to add a whitespace token, i would propose to try to decompose it by those points